packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -129,28 +293,123 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     }
   }
 
+  /**
+   * Sets the connected wallet
+   *
+   * @param wallet A wallet
+   */
   setWallet(wallet: Wallet | null) {
     this._wallet = wallet;
   }
 
-  setAccount(account: AccountInfo | null) {
-    this._account = account;
+  /**
+   * Sets the connected account
+   *
+   * `AccountInfo` type comes from a legacy wallet adapter plugin
+   * `StandardAccountInfo` type comes from AIP-62 standard compatible wallet when onAccountChange event is called
+   * `UserResponse<StandardAccountInfo>` type comes from AIP-62 standard compatible wallet on wallet connect
+   *
+   * @param account An account
+   */
+  setAccount(
+    account:
+      | AccountInfo
+      | StandardAccountInfo
+      | UserResponse<StandardAccountInfo>
+      | null
+  ): void {
+    if (!account) return;
+
+    // Check if wallet is of type AIP-62 standard
+    if (this._wallet?.isAIP62Standard) {
+      // Check if account is of type UserResponse<StandardAccountInfo> which means the `account`
+      // comes from the `connect` method
+      if ("status" in account) {
+        const connectStandardAccount =
+          account as UserResponse<StandardAccountInfo>;
+        if (connectStandardAccount.status === UserResponseStatus.REJECTED) {
+          this._connecting = false;
+          throw new WalletConnectionError("User has rejected the request")
+            .message;
+        }
+        // account is of type
+        this._account = {
+          address: connectStandardAccount.args.address.toString(),
```

Comment:
Is there a reason not to have it be `AccountAddress` or is that just for backwards compatibility?

Reply:
The wallet holds and returns it as `AccountAddress` (i.e `connectStandardAccount` is of type `AccountAddress`), as for the adapter we
1) want to keep backwards compatibility and 2) dont think it makes sense to returns to the dapp a `AccountAddress` type and have it resolves everything, the adapter should do it for them